src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,22 +43,46 @@
 
     function load(templateId) {
       $scope.loadInProgress = true;
-
-      return templates.getTemplateWithBody(templateId).then(function (result) {
+      return templates.getTemplate(templateId).then(function (template) {
+        switch (template.bodyType) {
+          case "empty":
+            $scope.isHtmlTemplate = !$rootScope.forceMsEditor;
+            break;
+          case "mseditor":
+            $scope.isHtmlTemplate = false;
+            break;
+          case "rawHtml":
+            $scope.isHtmlTemplate = true;
+            break;
+        }
+        return {
+          from_name: template.from_name,
+          from_email: template.from_email,
+          subject: template.subject,
+          id: template.id,
+          name: template.name,
+          body: $scope.isHtmlTemplate? templates.getTemplateBody(templateId) : null
+        };
+      }).then(function (result) {
         $scope.template.fromName = result.from_name;
         $scope.template.fromEmail = result.from_email;
         $scope.template.subject = result.subject;
         $scope.template.name = result.name;
         $scope.template.id = result.id;
-        $scope.template.content = result.body;
+        return result.body? result.body.then(function(body) {
+          return body.html;
+        }) : null;
+      }).then(function(html) {
+        $scope.template.content = html;
       }).finally(function () {
         $scope.loadInProgress = false;
       });
     }
 
-    function save() {
+    function saveTemplate() {
       $scope.saveInProgress = true;
-      return templates.save({
+      var saveTreatment = $scope.isHtmlTemplate? templates.saveHtml : templates.saveMSEditor;
```

Comment:
I would not differentiate between HTML and MSEditor at template service level, in my opinion, it is better to split saving Template and Template content. If your opinion is different we can discuss it.

Reply:
Please let me know when we can talk about it. Thanks @andresmoschini